waspc/src/Wasp/Generator/PackageJsonGenerator.hs

```diff
@@ -15,9 +15,9 @@ import qualified Data.Map as Map
 import Data.Maybe (fromMaybe)
 import qualified Data.Maybe as Maybe
 import Wasp.AppSpec (AppSpec)
-import qualified Wasp.AppSpec as AS
 import qualified Wasp.AppSpec.App as AS.App
```

Comment:
I was rather puzzled by AS.App and ASV. Is it not possible to import everything from one module? Can't you reexport the `App` type to `Valid`? Or alternatively use `getInvalidApp` in the places before validation, and keep `AS.getApp` to get the valid app?

Reply:
I see this causes a lot of changes to imports throughout the generator. I am having trouble understanding when I need to import from AS and when from ASV. Doesn't the generator deal only with validated app specs? I realize we won't do different types yet but a reexport would simplify matters, if possible.

Reply:
So, this is a bit tricky.

If we had the whole new intermediate representation that we know is valid, then we would import only that and work with just that.

But, we don't have that at the moment. We have the same old representation, which is AppSpec and all the data types that are used in it. And then we have AppSpec.Validation module, which does some validation on AppSpec.

I could reexport App from AppSpec.Valid, but what about the rest of IR? What about AppSpec.Page, or AppSpec.App.Auth, and so on? I would need to reexport all of those, which becomes weird then, better to use the original structure.

I could reexport `validate` stuff from AppSpec maybe -> but then I would get circular dependencies.
I could avoid them by differently organizing things, extracting some stuff, but I think it would just further complicate things and not bring us much benefits.

So the approach I chose is to say that we have this AppSpec.Valid module, that knows how to do validation, and it allows you to do some stuff with AppSpec that makes sense only in the context of validated AppSpec. I wish I could have bought that on type level, but as was discussed, we gave up on that for now. Still, I like the idea that `getApp` is coming from `AppSpec.Valid` module -> it is connected to validation since it relies on it, it knows about concept of validation. While the AppSpec itself doesn't, and that is ok.
Also, this keeps `getApp` close to validation logic, and since it relies on that code, I also liked that -> it is like that person changing that code will also notice `getApp` and take it into account.

At the end it really iches me that we are not enforcing this in some better way on the type level, but we did a lot of consideration and it made sense to be practical at the moment.

What do you think, does this make sense?

Reply:
I added some comments on these functions to be explicit about the assumptions.

Reply:
The imports appear to look simpler, so I think we succeeded!